### PR TITLE
Adds JobTemplate resource

### PR DESCRIPTION
**Pull request by shangabl**
This commit adds the CloudFormation resource AWS::IoT::JobTemplate including the schema and handlers

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-iot-jobtemplate/src/main/java/software/amazon/iot/jobtemplate/Translator.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+public class Translator {
+
+    static software.amazon.awssdk.services.iot.model.AbortConfig getAbortConfig(ResourceModel model) {
+        if(model == null || model.getAbortConfig() == null || model.getAbortConfig().getCriteriaList().isEmpty()) {
+            return null;
+        }
+        List<software.amazon.iot.jobtemplate.AbortCriteria> listModelCriteria = model.getAbortConfig().getCriteriaList();
+
+        List<software.amazon.awssdk.services.iot.model.AbortCriteria> listSdkCriteria = listModelCriteria.stream().map( modelCriteria -> {
```
Comment (BenChen-AWS) [inline aws-iot-jobtemplate/src/main/java/software/amazon/iot/jobtemplate/Translator.java:22]: Can we use a unified name for the term? I prefer to put List at the end of string like the one on line 39.

File: aws-iot-jobtemplate/src/test/java/software/amazon/iot/jobtemplate/CreateHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+        assertThat(response.getCallbackContext()).isNull();
+        assertThat(response.getCallbackDelaySeconds()).isEqualTo(0);
+        assertThat(response.getResourceModels()).isNull();
+        assertThat(response.getMessage()).isNull();
+        assertThat(response.getErrorCode()).isNull();
+        assertThat(response.getResourceModel()).isEqualTo(expectedModel);
+    }
+
+    @Test
+    public void handleRequest_ResourceAlreadyExists() {
```
Comment (BenChen-AWS) [inline aws-iot-jobtemplate/src/test/java/software/amazon/iot/jobtemplate/CreateHandlerTest.java:84]: Make method name consistent with testing exception. Same for other unit tests.

File: aws-iot-jobtemplate/src/test/java/software/amazon/iot/jobtemplate/HandlerTestBase.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+    CreateJobTemplateResponse getCreateResponse() {
+        return CreateJobTemplateResponse.builder()
+                .jobTemplateArn(JOB_TEMPLATE_ARN)
+                .jobTemplateId(JOB_TEMPLATE_ID)
+                .build();
+    }
+
+    DescribeJobTemplateResponse getDescribeResponse() {
+        final software.amazon.awssdk.services.iot.model.AbortCriteria c = software.amazon.awssdk.services.iot.model.AbortCriteria.builder()
```
Comment (BenChen-AWS) [inline aws-iot-jobtemplate/src/test/java/software/amazon/iot/jobtemplate/HandlerTestBase.java:62]: Use meaningful naming.

File: aws-iot-jobtemplate/src/main/java/software/amazon/iot/jobtemplate/CreateHandler.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            throw new CfnServiceInternalErrorException(OPERATION, e);
+        } catch (final ThrottlingException e) {
+            throw new CfnThrottlingException(OPERATION, e);
+        } catch (final ServiceUnavailableException e) {
+            throw new CfnGeneralServiceException(OPERATION, e);
+        } catch (final UnauthorizedException e) {
+            throw new CfnAccessDeniedException(OPERATION, e);
+        } catch (final Exception e) {
+            throw new CfnInternalFailureException(e);
+        }
```
Comment (BenChen-AWS) [inline aws-iot-jobtemplate/src/main/java/software/amazon/iot/jobtemplate/CreateHandler.java:108]: Add unit test to ensure we cover unknown exception.

**Review by BenChen-AWS — COMMENTED** [review]
Add comments.

**Review by BenChen-AWS — APPROVED** [review]
Looks good to me.

**Review by HarshGandhi-AWS — APPROVED** [review]

**Review by HarshGandhi-AWS — APPROVED** [review]

**Pull request merged**